### DISCUSSION TITLE
Question about how to get the header file

### DISCUSSION THREAD

**Discussion by Goodweatherz** [kschan0214/sepia#75] [Q&A] [answered]
Hi, I currently possess magnitude and phase maps in NIFTI format. However, I find myself in need of the corresponding header files for these images. When I tried to use 'Get header info' in the 'Utility' tab of the GUI to generate header file, I got 'No TE file is identified.' errors. 
Could you kindly provide instructions or insights on how to extract the header files from the existing magnitude and phase maps in NIFTI format? or could you tell me what the format and content of TE file looks like?
Thank you for your time.

**Comment by kschan0214** [comment, marked as answer]
Hi, 

Unfortunately, the echo time information is not commonly stored in the NIFTI header. There are several ways to extract the TE info from other file formats:

### Option 1: JSON file(s) from dcm2niix
If the NIFTI files were generated from dcm2niix, there should be JSON sidecar files accompanied by the NIFTI. You can select all JSON files associated with the (multi-echo) data using the 'and TE file(s)' under the 'Get header info' utility. 

Example JSON files:
![](https://sepia-documentation.readthedocs.io/en/latest/_images/bids_directory.png)

Select all JSON files here
<img width="983" alt="Screenshot 2023-12-12 at 23 26 34" src="https://github.com/kschan0214/sepia/assets/39571502/5ed485b3-9ebf-41d0-8c1b-d6229857505a">

### Option 2: a single JSON file from dicm2nii
If dicm2nii was used to create the NIFTI, all TEs will be stored in a single JSON file instead.

### Option 3: A text file
If MRIConvert was used to create the NIFTI, the TE info is stored in a separate text file. 

### Option 4: A Matlab's .mat file
The '.mat' file should have the TE info stored under the variable name 'TE' in a unit of second. It might be useful if there is already a sepia_header.mat available (e.g. from another dataset with the same TE)

### Option 5: User input via GUI
If none of the above options works, you can also manually enter the echo time (see example below)
<img width="1003" alt="Screenshot 2023-12-12 at 23 40 48 1" src="https://github.com/kschan0214/sepia/assets/39571502/4f3a0edd-880c-46bd-86f8-45a95ec6efa0">

### Option 6: User input via command windows
As a last resort you can also manually create the sepia_header.mat file using the command window. Here is an example:
```
B0 = 3;                 % magnetic field strength, in Tesla
TE = [te1,te2,te3,te4]; % echo time for each GRE image, in second
save('sepia_header.mat','B0','TE');
```
Just replace 'te1', te2', etc. with the real values. You may find more info on the documentation webpage: https://sepia-documentation.readthedocs.io/en/latest/getting_started/Sepia-header.html

Let me know if this helps!

Kwok

**Reply by Goodweatherz** [reply]
Thank you very much!
